controller/app/validators/key_content_validator.rb

```diff
@@ -0,0 +1,15 @@
+class KeyContentValidator < ActiveModel::Validator
+   def validate(record)
+     key = Net::SSH::KeyFactory.load_data_public_key(record.type + " " + record.content)
```

Comment:
We should probably add that in addition to the existing rescue as it's catching other error types already. Ruby is not my area of expertise, so I'm not too familiar with it's exception handling.